lib/sd_notify.rb

```diff
@@ -104,21 +106,16 @@ def self.watchdog?
   def self.notify(state, unset_env=false)
     sock = ENV["NOTIFY_SOCKET"]
 
-    return nil if !sock
+    return nil unless sock
 
     ENV.delete("NOTIFY_SOCKET") if unset_env
 
-    connected = false
-
     begin
-      sock = Addrinfo.unix(sock, :DGRAM).connect
-      connected = true
-      sock.close_on_exec = true
```

Comment:
TIL that Ruby 2.0.0+ automatically sets this flag. Perhaps this should be stated in the relevant commit description, for future reference.

Reply:
:thinking: I'm not actually sure I get in which context this flag was initially useful. Was it to close the socket if the error handler is forking a new process and does `exec` behind ?

Do you have a specific test case in mind?

Reply:
Not sure either, but I guess it's the sane thing to do (not sure why CLOEXEC isn't the default when opening new sockets). The original implementation also [sets it](https://github.com/systemd/systemd/blob/v244/src/libsystemd/sd-daemon/sd-daemon.c#L479).

Also, from man pages of `sd_listen_fds(3)`:

> This function call will set the FD_CLOEXEC flag for all passed file descriptors to avoid further inheritance to children of the calling process

I guess you just don't want other threads spawned from the main process to mess with this fd.

Reply:
On a second thought, maybe we could leave the explicit set of `close_on_exec = true` so that we don't depend on Ruby's defaults and that the code is more clear?

As for the test case, if we make it explicit I don't think it's worth it.

Reply:
Yeah, why not.
At least it will educate people :)

Reply:
Updated, with "kind of" meaningful commit message.